### ISSUE TITLE
no-negated-condition and if/else statements with three or more branches

### ISSUE THREAD

**Issue by reywood** [eslint/eslint#9913]
**Tell us about your environment**

* **ESLint Version:** 4.16.0
* **Node Version:** 4.4.7
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

none

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "root": true,
    "rules": {
        "no-negated-condition": 2
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// ESLint does not like the `else if` conditional here

function doSomething(foo, bar, callback) {
    if (foo !== 'expected value for foo') {
        callback(new Error('Unexpected value for foo'));
    } else if (bar !== 'expected value for bar') {
        callback(new Error('Unexpected value for bar'));
    } else {
        callback(null, 'foo and bar are good');
    }
}

// ESLint likes this, but the logic suffers

function doSomething(foo, bar, callback) {
    if (foo !== 'expected value for foo') {
        callback(new Error('Unexpected value for foo'));
    } else if (bar === 'expected value for bar') {
        callback(null, 'foo and bar are good');
    } else {
        callback(new Error('Unexpected value for bar'));
    }
}
```

```bash
eslint
```

**What did you expect to happen?**

The intended effect of the above example is that two variables are checked, and if they don't match expected values, then an informative error is returned to the callback. If both values are good, a successful response is returned.

This is a somewhat contrived example, but the intent is to illustrate that the value of the `no-negated-condition` rule is not so clear for an `if`/`else if`/`else` statement. At least, the value is not nearly as clear as it is for a simple `if`/`else` statement where reversing the contents of the `if` and `else` blocks does not really distort the semantic flow.

The above code example could be rewritten as follows and ESLint will no longer complain.

```js
function doSomething(foo, bar, callback) {
    if (foo !== 'expected value for foo') {
        callback(new Error('Unexpected value for foo'));
        return;
    }
    if (bar !== 'expected value for bar') {
        callback(new Error('Unexpected value for bar'));
        return;
    }
    callback(null, 'foo and bar are good');
}
```

However, it could be argued that this simply skirts the intention of the rule without improving the code.

Should this rule apply to if/else statements with three or more branches?

**Comment by platinumazure** [comment]
Hi @reywood, thanks for the issue! Sorry for the delayed response.

This is an interesting use case. I've looked at the rule source and our unit test cases, and it looks like we simply haven't considered how the rule would work with an if statement with 3 or more parts.

I honestly don't know whether to consider this a bug or an enhancement request at this point.

**Comment by platinumazure** [comment]
@alberto Seems you might have originally written the rule. Do you have any thoughts here?

**Comment by reywood** [comment]
😢
